lib/std/reflection.nim

```diff
@@ -0,0 +1,80 @@
+##[
+This module implements some reflection API's.
+It is supported in c, cpp, js backends.
+
+Experimental API, subject to change.
+]##
+
+runnableExamples:
+  # `getProcname`
+  proc fn1(a: int, b = 'x') =
+    assert getProcname() == "fn1"
+    static: assert getProcname(withType = true) == "fn1: proc (a: int; b: char)"
+  fn1(1)
+
+  iterator fn3[T](a: T): string =
+    static: doAssert getProcname() == "fn3"
+    yield getProcname(withType = true)
+
+  from std/sequtils import toSeq
+  assert toSeq(fn3(1.5)) == @["fn3: proc (a: float64): string"]
+
+runnableExamples:
+  # `getExportcName`
+  from std/strutils import contains
+  proc fn1 =
+    assert "fn1" in getExportcName() # implementation defined, e.g. `fn1_reflection95examples49_1`
+  fn1()
+
+  proc fn2: auto {.exportc.} = getExportcName()
+  assert fn2() == "fn2"
+
+  iterator fn3[T](a: T): string {.closure.} =
+    yield getExportcName()
+
+  from std/sequtils import toSeq
+  let s1 = toSeq(fn3(1.5))[0]
+  let s2 = toSeq(fn3(1))[0]
+  assert "fn3" in s1
+  assert s2 != s1
+
+import std/macros
+
+template getExportcName*(): string =
```

Comment:
`std/stackframes` can use this implementation too.

Reply:
I've now instead deprecated `stackframes.procName`, it was poorly named and std/reflection is a better home for this